### PR TITLE
config: Randomize MAC Addresses

### PR DESCRIPTION
**Pull request by RamLavi**
This PR is changing the MAC Addresses to be configured randomly in case not set by user.
This change will allow running this checkup multiple times in parallel in the future.

### CODE REVIEW

**Comment by RamLavi** [comment]
example output: MACs changed to
```
2023/02/28 14:11:17 "trafficGeneratorEastMacAddress": "50:13:96:23:66:01"
2023/02/28 14:11:17 "trafficGeneratorWestMacAddress": "50:1c:9e:d2:35:02"
2023/02/28 14:11:17 "DPDKEastMacAddress": "60:b1:65:c8:15:01"
2023/02/28 14:11:17 "DPDKWestMacAddress": "60:e7:fc:93:97:02"
```

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns make TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest test/e2e 
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           -e RUNTIME_CLASS_NAME= \
           -e TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest \
           -e VM_CONTAINER_DISK_IMAGE_URL= \
           docker.io/library/golang:1.19.4 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1677593451

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:63
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
  [FAILED] in [It] - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:89 @ 02/28/23 14:14:11.856
• [FAILED] [200.873 seconds]
Execute the checkup Job [It] should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79

  [FAILED] checkup failed: {
  	"spec.param.networkAttachmentDefinitionName": "intel-dpdk-network-1",
  	"spec.param.testDuration": "1m",
  	"spec.param.trafficGeneratorImage": "quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest",
  	"spec.param.trafficGeneratorRuntimeClassName": "performance-performance-zeus10",
  	"spec.param.verbose": "true",
  	"spec.param.vmContainerDiskImage": "",
  	"spec.timeout": "10m",
  	"status.completionTimestamp": "2023-02-28T14:14:07Z",
  	"status.failureReason": "detected 4219379968.000000 Bps Drop Rate on the traffic generator's side",
  	"status.result.DPDKRxPacketDrops": "39792340",
  	"status.result.DPDKTxPacketDrops": "0",
  	"status.result.DPDKVMNode": "zeus10.lab.eng.tlv2.redhat.com",
  	"status.result.trafficGeneratorInErrorPackets": "0",
  	"status.result.trafficGeneratorMaxDropRate": "4219379968.000000",
  	"status.result.trafficGeneratorNode": "zeus10.lab.eng.tlv2.redhat.com",
  	"status.result.trafficGeneratorOutputErrorPackets": "0",
  	"status.startTimestamp": "2023-02-28T14:11:17Z",
  	"status.succeeded": "false"
  }
  In [It] at: /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:89 @ 02/28/23 14:14:11.856
------------------------------

Summarizing 1 Failure:
  [FAIL] Execute the checkup Job [It] should complete successfully
  /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:89

Ran 1 of 1 Specs in 200.877 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestKubevirtDpdkCheckup (200.88s)
FAIL
FAIL	github.com/kiagnose/kubevirt-dpdk-checkup/tests	200.911s
FAIL
make: *** [Makefile:59: test/e2e] Error 1

```

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi.
The VM latency check has an [implementation](https://github.com/kiagnose/kiagnose/blob/main/checkups/kubevirt-vm-latency/vmlatency/internal/vmi/spec.go#L229) of such functionality.

Please consider using it here as well.

**Comment by RamLavi** [comment]
> Could you please change "MACS" to "MAC addresses" in the PR title and description?

DONE

**Comment by orelmisan** [comment]
BTW, IMO the PR title is misleading, the MAC addresses are currently hard-coded.
This PR makes them random.

**Comment by RamLavi** [comment]
> BTW, IMO the PR title is misleading, the MAC addresses are currently hard-coded. This PR makes them random.

is it better now?

**Comment by RamLavi** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/2fd6f98b8c4e0ada4e1e1a640e97fece2347e419..21e75ce4ac1f0c6ddaca1f5a43dfa5b259d58d44): Rebase

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns make TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest test/e2e 
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           -e RUNTIME_CLASS_NAME= \
           -e TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest \
           -e VM_CONTAINER_DISK_IMAGE_URL= \
           docker.io/library/golang:1.19.4 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1677678980

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:63
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
  [FAILED] in [It] - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:89 @ 03/01/23 14:00:05.949
• [FAILED] [225.795 seconds]
Execute the checkup Job [It] should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79

  [FAILED] checkup failed: {
  	"spec.param.networkAttachmentDefinitionName": "intel-dpdk-network-1",
  	"spec.param.testDuration": "1m",
  	"spec.param.trafficGeneratorImage": "quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:latest",
  	"spec.param.trafficGeneratorRuntimeClassName": "performance-performance-zeus10",
  	"spec.param.verbose": "true",
  	"spec.param.vmContainerDiskImage": "",
  	"spec.timeout": "10m",
  	"status.completionTimestamp": "2023-03-01T14:00:00Z",
  	"status.failureReason": "detected Error Packets on the traffic generator's side: Oerrors 0 Ierrors 376",
  	"status.result.DPDKRxPacketDrops": "40664454",
  	"status.result.DPDKRxTestPackets": "562558115",
  	"status.result.DPDKTxPacketDrops": "0",
  	"status.result.DPDKVMNode": "zeus10.lab.eng.tlv2.redhat.com",
  	"status.result.trafficGeneratorInErrorPackets": "376",
  	"status.result.trafficGeneratorNode": "zeus10.lab.eng.tlv2.redhat.com",
  	"status.result.trafficGeneratorOutputErrorPackets": "0",
  	"status.result.trafficGeneratorTxPackets": "840000000",
  	"status.startTimestamp": "2023-03-01T13:56:31Z",
  	"status.succeeded": "false"
  }
  In [It] at: /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:89 @ 03/01/23 14:00:05.949
------------------------------

Summarizing 1 Failure:
  [FAIL] Execute the checkup Job [It] should complete successfully
  /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:89

Ran 1 of 1 Specs in 225.800 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestKubevirtDpdkCheckup (225.80s)
FAIL
FAIL	github.com/kiagnose/kubevirt-dpdk-checkup/tests	225.830s
FAIL
make: *** [Makefile:59: test/e2e] Error 1
```

**Comment by RamLavi** [comment]
Updated README info

File: README.md [hunk 1]
```diff
@@ -68,23 +68,23 @@ roleRef:
...
+| spec.param.trafficGeneratorImage            | Traffic generator's container image                                                                               | False        | Defaults to the U/S image https://quay.io/repository/kiagnose/kubevirt-dpdk-checkup-traffic-gen:latest |
+| spec.param.trafficGeneratorNodeSelector     | Node Name on which the traffic generator Pod will be scheduled to                                                 | False        | Assumed to be configured to Nodes that allow DPDK traffic                                              |
+| spec.param.trafficGeneratorPacketsPerSecond | Amount of packets per second in Millions                                                                          | False        | Defaults to 14                                                                                         |
+| spec.param.trafficGeneratorEastMacAddress   | MAC address of the NIC connected to the traffic generator pod/VM                                                  | False        | Defaults to a random MAC with prefix 0x50 and suffix 0x01                                              |
```
Comment (orelmisan) [inline README.md:79]: I suggest to change the remarks format to:
Defaults to a random MAC address of the form: 50:xx:xx:xx:xx:01

Comment (RamLavi) [reply README.md]: DONE

**Review by orelmisan — APPROVED** [review]
Thank you for the changes

**Pull request merged**